src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2632,6 +2632,40 @@ _ostree_deployment_set_bootconfig_from_kargs (OstreeDeployment *deployment,
     }
 }
 
+// Perform some basic static analysis and emit warnings for things
+// that are likely to fail later.  This function only returns
+// a hard error if something unexpected (e.g. I/O error) occurs.
+static gboolean
+lint_deployment_fs (OstreeSysroot     *self,
+                    OstreeDeployment  *deployment,
+                    int                deployment_dfd,
+                    GCancellable      *cancellable,
+                    GError           **error)
+{
+  g_auto(GLnxDirFdIterator) dfd_iter = { 0, };
+  glnx_autofd int dest_dfd = -1;
+  gboolean exists;
+
+  if (!ot_dfd_iter_init_allow_noent (deployment_dfd, "var", &dfd_iter, &exists, error))
+    return FALSE;
+  while (exists)
+    {
+
+      struct dirent *dent;
+
+      if (!glnx_dirfd_iterator_next_dent (&dfd_iter, &dent, cancellable, error))
+        return FALSE;
+      if (dent == NULL)
+        break;
+
+      ot_journal_print (LOG_WARNING, "Deploying commit %s which contains content in /var/%s that will be ignored.",
```

Comment:
Hmm, wouldn't `g_warning` be a better fit here? E.g. in cosa for example this will run in supermin and we don't collect journal logs there. Or in Anaconda, it'd be harder to miss if it shows up inline with the rest of the Anaconda output/logs.

Reply:
I think logging to the journal as a general rule has more elegance.  But...yes you're right, we have issues where we're not always run under systemd.

I used `fprintf (stderr, ...)` and not `g_warning()` because we use `G_DEBUG=fatal-warnings` in some cases.

It's funny: I changed the code to do that but didn't change the test which grepped the journal...and the test still passed, because of course the kola exttests are logging to the journal.

Reply:
Could `ot_journal_print` just write to stderr if the journal isn't enabled?

Reply:
Ultimately...a big (huge, really) benefit of systemd over SysV is that actually just calling `printf/fprintf(stderr)` Just Works too.  So I think this is fine.  We could add such infrastructure, but it's not clear in this case it'd really be better.